anndata/experimental/merge.py

```diff
@@ -0,0 +1,615 @@
+        If False the and if a file already exists it will raise an error,
+        otherwise it will overwrite.
+    max_loaded_sparse_elems
+        The maximum number of elements to load in memory when concatenating
+        sparse arrays. Note that this number also includes the empty entries.
+        Set to 100m by default meaning roughly 400mb will be loaded
+        to memory at simultaneously.
+    axis
+        Which axis to concatenate along.
+    join
+        How to align values when concatenating. If "outer", the union of the other axis
+        is taken. If "inner", the intersection. See :doc:`concatenation <../concatenation>`
+        for more.
+    merge
+        How elements not aligned to the axis being concatenated along are selected.
+        Currently implemented strategies include:
+
+        * `None`: No elements are kept.
+        * `"same"`: Elements that are the same in each of the objects.
+        * `"unique"`: Elements for which there is only one possible value.
+        * `"first"`: The first element seen at each from each position.
+        * `"only"`: Elements that show up in only one of the objects.
+    uns_merge
+        How the elements of `.uns` are selected. Uses the same set of strategies as
+        the `merge` argument, except applied recursively.
+    label
+        Column in axis annotation (i.e. `.obs` or `.var`) to place batch information in.
+        If it's None, no column is added.
+    keys
+        Names for each object being added. These values are used for column values for
+        `label` or appended to the index if `index_unique` is not `None`. Defaults to
+        incrementing integer labels.
+    index_unique
+        Whether to make the index unique by using the keys. If provided, this
+        is the delimeter between "{orig_idx}{index_unique}{key}". When `None`,
+        the original indices are kept.
+    fill_value
+        When `join="outer"`, this is the value that will be used to fill the introduced
+        indices. By default, sparse arrays are padded with zeros, while dense arrays and
+        DataFrames are padded with missing values.
+    pairwise
+        Whether pairwise elements along the concatenated dimension should be included.
+        This is False by default, since the resulting arrays are often not meaningful.
+
+    Notes
+    -----
+
+    .. warning::
+
+        If you use `join='outer'` this fills 0s for sparse data when
+        variables are absent in a batch. Use this with care. Dense data is
+        filled with `NaN`.
+    """
+    # Argument normalization
+    if pairwise:
+        raise NotImplementedError("pairwise concatenation not yet implemented")
+    if join != "inner":
+        raise NotImplementedError("only inner join is currently supported")
+
+    merge = resolve_merge_strategy(merge)
+    uns_merge = resolve_merge_strategy(uns_merge)
+    if len(in_files) <= 1:
+        raise ValueError("Must pass at least two files to concatenate.")
```

Comment:
Is there a reason to not just do sth. like this here? (needs `shutil` import)

```suggestion
    if len(in_files) <= 1:
        if len(in_files) == 1:
            if not overwrite and Path(out_file).is_file():
                raise FileExistsError(f"File “{out_file}” already exists and `overwrite` is set to False")
            shutil.copy2(in_files[0], out_file)
        return
```

I imagine someone might use this on collections of files of varying length.